services/controllerhost/event_handlers.go

```diff
@@ -99,6 +99,16 @@ type (
 		storeIDs []string
 	}
 
+	// StoreRemoteExtentReplicatorDownEvent is triggered
+	// for an extent when the primary store host that's
+	// responsible for replicating that extent from origin
+	// zone is down
+	StoreRemoteExtentReplicatorDownEvent struct {
```

Comment:
yeah.. I didn't like that name either.. `RemoteExtentPrimaryStore` makes me think its a store in a remote DC.. Don't like both the names :) But I will go with yours.